lms/djangoapps/certificates/apis/v0/permissions.py

```diff
@@ -0,0 +1,29 @@
+"""
+This module provides a custom DRF Permission class for supporting the course certificates
+to Admin users and users whom they belongs to.
+"""
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from rest_framework.permissions import BasePermission
+
+from openedx.core.djangoapps.user_api.accounts.serializers import get_profile_visibility
+
+
+class IsAdminOrSelfUser(BasePermission):
+    """
+    Method that will ensure whether the requesting user is staff or
+    the user whom the certificate belongs to
+    """
+    def has_permission(self, request, view):
+        user = request.user
+        requested_profile_username = view.kwargs.get('username')
+
+        if user.is_staff or user.username == requested_profile_username:
+            return True
+
+        user = User.objects.get(username=requested_profile_username)
+        configuration = settings.ACCOUNT_VISIBILITY_CONFIGURATION
+
+        account_privacy = get_profile_visibility(user.profile, user, configuration)
+        return account_privacy == 'all_users'
```

Comment:
1. I imagine that are a lot more errors from people not being able to see their own certificates, so I would land that PR more quickly and then iterate.
2. It is not use a top-level configuration to share the profile, but also this permission from the Profile page:
<img width="872" alt="Screen Shot 2020-06-01 at 5 45 03 PM" src="https://user-images.githubusercontent.com/14576445/83458645-d492d700-a430-11ea-9b7e-5185f2df4768.png">

Reply:
@robrap this drop-down ultimately hits the preference api resulting in updation/creation of preference.
`get_profile_visibility` is already taking care of it.

Reply:
I was pretty sure that there is **both** profile visibility as well as field level visibility added into the newer profile.  You'll notice in the profile page that there is a "Who can see this" next to several different fields, and you can have different visibility for each.  I don't know how the check is done, and if it is just a different configuration, but `ACCOUNT_VISIBILITY_CONFIGURATION` seems like it is too global.

You can test what I am saying by setting the visibility for Name or Country to "Everyone", and leaving Certificates visibility to "Just me", and see if this current code is incorrectly providing access or not.

Reply:
@uzairr agree with Robert, could you please check if we can use field level visibility configuration here?